runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -141,26 +164,49 @@ private void translateKinesisInput(
     RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
 
     String stream;
+    FlinkLyftKinesisConsumer<WindowedValue<byte[]>> source;
+
     Properties properties = new Properties();
     ObjectMapper mapper = new ObjectMapper();
     try {
-      Map<String, Object> params =
-          mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
+      JsonNode params = mapper.readTree(pTransform.getSpec().getPayload().toByteArray());
 
       Preconditions.checkNotNull(
-          stream = (String) params.get("stream"), "'stream' needs to be set");
-      Map<?, ?> consumerProps = (Map) params.get("properties");
+          stream = params.path("stream").textValue(), "'stream' needs to be set");
+
+      Map<?, ?> consumerProps = mapper.convertValue(params.path("properties"), Map.class);
       Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
       properties.putAll(consumerProps);
+
+      String encoding = BYTES_ENCODING;
+      if (params.hasNonNull("encoding")) {
+        encoding = params.get("encoding").asText();
+      }
+
+      switch (encoding) {
+        case BYTES_ENCODING:
+          source =
+              FlinkLyftKinesisConsumer.create(
+                  stream, new KinesisByteArrayWindowedValueSchema(), properties);
+          break;
+        case LYFT_KINESIS_EVENT_ENCODING:
```

Comment:
Minor: Lyft has more than one schema and I don't know if this is the most common one. We cannot solve this in a generic way, but maybe be more specific what it is "LYFT_ZLIB_BASE64_JSON_ENCODING" ?

Reply:
Good call.